core/server/auth/oauth.js

```diff
@@ -85,6 +124,13 @@ oauth = {
         // access token on behalf of the user who authorized the code.
         oauthServer.exchange(oauth2orize.exchange.refreshToken({userProperty: 'client'},
             exchangeRefreshToken));
+
+        /**
+         * We forward the authorization code to Ghost.org, so no need to call exchange.authorizationCode
+         * exchange.authorizationCode wraps the req and res object, which is used by passport!
+         * See exchangeAuthorizationCode!
+         */
+        oauthServer.exchange('authorization_code', exchangeAuthorizationCode);
```

Comment:
`exchange.authorizationCode` takes over client extraction which is missing in the code above.